src/core/util/jsonHash.js

```diff
@@ -0,0 +1,22 @@
+const xxHash = require('xxhashjs')
+const stringify = require('json-stringify-deterministic')
+
+/**
+ * Module JSON Hash.
+ * 
+ * @module jsonHash
+ */
+
+/**
+ * The json object is stringified by sorting the keys, removing spaces and new lines. It is then passed to a function to
+ * calculate the hash code.
+ * 
+ * @param {*} jsonObject - A json object.
+ * @returns a string representing 64-bit hash
+ */
+module.exports = function jsonHash(jsonObject) {
+    const canonicalJsonObject = stringify(jsonObject)
+    return xxHash.h64(canonicalJsonObject, 0).toString(16)
```

Comment:
for the same rules it will give the same hash, right? There is no salt or some 'randomness' there which will lead to a different hash each time, right?

Reply:
After additional thoughts --> In general, I think a talent a new instance of a talent should not receive any events subscribed by an older / previous instance of it even it has the same rule-set.

At the end, the main problem is, that the uuid of a talent is not completely unique. What we could also do is to make the uuid if a talent instance specific. 'name'+ uuid (which changes for every instance). That could make it much easier and you don't have to calculate a hash over the rule set or the discovery reply. Maybe we can check this again?

Reply:
> make the uuid of a talent instance specific
I think that's a very good approach!

Reply:
> After additional thoughts --> In general, I think a talent a new instance of a talent should not receive any events subscribed by an older / previous instance of it even it has the same rule-set.
> 
> At the end, the main problem is, that the uuid of a talent is not completely unique. What we could also do is to make the uuid if a talent instance specific. 'name'+ uuid (which changes for every instance). That could make it much easier and you don't have to calculate a hash over the rule set or the discovery reply. Maybe we can check this again?

Wouldn't that break shared subscriptions and thus scaling: currently all instances of a Talent subscribe for events with the topic: $share/<talent_id>/talent/<talent_id>/events

if every instance has a different talent id, the MQTT shared subscriptions will not work.